### ISSUE TITLE
Stocks not working correctly

### ISSUE THREAD

**Issue by muff1nman** [nicho92/MtgDesktopCompanion#14]
After inserting a stock, it doesn't return after reloading the panel.

Looks to be a discrepancy between the `idmc` column on insert and query.

On insert the following logic is used:

```
idmc = state.getMagicCard().getId()
```

and on query the following is used:

```
idmc == IDGenerator.generate(mc)
```

**Comment by muff1nman** [comment]
Looking at the Mysql dao, it looks like using IDGenerate is what is intended here?

**Comment by nicho92** [comment]
Hi, 
Thanks a lot for your feedback .. i was alone testing my app :)

IDGenerate allow me to have a "unique" id for cards (especially when you use other providers with their own ids).
Daos must use this function and never save "mc.getID".

**Comment by nicho92** [comment]
fixed ebd7809
